client/my-sites/plugins/plugin-management-v2/plugin-details-v2/index.tsx

```diff
@@ -0,0 +1,119 @@
+import { useTranslate } from 'i18n-calypso';
+import { useSelector } from 'react-redux';
+import QueryEligibility from 'calypso/components/data/query-atat-eligibility';
+import QueryJetpackPlugins from 'calypso/components/data/query-jetpack-plugins';
+import QueryProductsList from 'calypso/components/data/query-products-list';
+import QuerySiteFeatures from 'calypso/components/data/query-site-features';
+import FixedNavigationHeader from 'calypso/components/fixed-navigation-header';
+import PluginNotices from 'calypso/my-sites/plugins/notices';
+import PluginDetailsBody from 'calypso/my-sites/plugins/plugin-details-body';
+import PluginDetailsHeader from 'calypso/my-sites/plugins/plugin-details-header';
+import PluginAvailableOnSitesList from 'calypso/my-sites/plugins/plugin-management-v2/plugin-details-v2/plugin-available-on-sites-list';
+import SitesWithInstalledPluginsList from 'calypso/my-sites/plugins/plugin-management-v2/plugin-details-v2/sites-with-installed-plugin-list';
+import { siteObjectsToSiteIds } from 'calypso/my-sites/plugins/utils';
+import {
+	getSiteObjectsWithPlugin,
+	getSiteObjectsWithoutPlugin,
+} from 'calypso/state/plugins/installed/selectors';
+import { isFetching as isWporgPluginFetchingSelector } from 'calypso/state/plugins/wporg/selectors';
+import getSelectedOrAllSites from 'calypso/state/selectors/get-selected-or-all-sites';
+import type { Plugin } from '../types';
+import type { SiteData } from 'calypso/state/ui/selectors/site-data';
+import type { ReactElement } from 'react';
+
+import './style.scss';
+
+interface Props {
+	selectedSite: SiteData;
+	pluginSlug: string;
+	fullPlugin: Plugin;
+	sitesWithPlugins: Array< SiteData >;
+	showPlaceholder: boolean;
+	isMarketplaceProduct: boolean;
+	isWpcom: boolean;
+}
+
+export default function PluginDetailsV2( {
+	selectedSite,
+	pluginSlug,
+	fullPlugin,
+	sitesWithPlugins,
+	showPlaceholder,
+	isMarketplaceProduct,
+	isWpcom,
+}: Props ): ReactElement {
+	const translate = useTranslate();
+
+	const siteIds: any = [ ...new Set( siteObjectsToSiteIds( sitesWithPlugins ) ) ];
+	const sitesWithPlugin = useSelector( ( state ) =>
+		getSiteObjectsWithPlugin( state, siteIds, pluginSlug )
+	);
```

Comment:
I noticed that `fullPlugin.sites` has the list of sites that have the plugin installed. Would it make sense to use that list instead of computing it?

Reply:
I checked it, we don't get the site information in `fullPlugin.sites`

<img width="806" alt="Screenshot 2022-08-25 at 3 49 57 PM" src="https://user-images.githubusercontent.com/10586875/186640658-21202b84-2a4e-4329-88d1-605d838e11c9.png">